es_qat_wrapper/8.0.0/lucene-8.2.0/lucene/core/src/test/org/apache/lucene/codecs/compressing/AbstractTestCompressionMode.java

```diff
@@ -0,0 +1,215 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.lucene.codecs.compressing;
+
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import com.carrotsearch.randomizedtesting.generators.RandomNumbers;
+import org.apache.lucene.store.ByteArrayDataInput;
+import org.apache.lucene.store.ByteArrayDataOutput;
+import org.apache.lucene.util.ArrayUtil;
+import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.LuceneTestCase;
+import org.apache.lucene.util.TestRuleLimitSysouts;
+import org.apache.lucene.util.TestUtil;
+
+@TestRuleLimitSysouts.Limit(bytes = 204200)
+public abstract class AbstractTestCompressionMode extends LuceneTestCase {
+
+  CompressionMode mode;
+
+  static byte[] randomArray() {
+    final int max = random().nextBoolean()
+        ? random().nextInt(4)
+        : random().nextInt(255);
+    final int length = random().nextBoolean()
+        ? random().nextInt(20)
+        : random().nextInt(192 * 1024);
+    return randomArray(length, max);
+  }
+
+  static byte[] randomArray(int length, int max) {
+    final byte[] arr = new byte[length];
+    for (int i = 0; i < arr.length; ++i) {
+      arr[i] = (byte) RandomNumbers.randomIntBetween(random(), 0, max);
+    }
+    return arr;
+  }
+
+  byte[] compress(byte[] decompressed, int off, int len) throws IOException {
+    Compressor compressor = mode.newCompressor();
+    return compress(compressor, decompressed, off, len);
+  }
+
+  static byte[] compress(Compressor compressor, byte[] decompressed, int off, int len) throws IOException {
+    byte[] compressed = new byte[len * 2 + 1000]; // should be enough
+    ByteArrayDataOutput out = new ByteArrayDataOutput(compressed);
+    compressor.compress(decompressed, off, len, out);
+    final int compressedLen = out.getPosition();
+    return ArrayUtil.copyOfSubArray(compressed, 0, compressedLen);
+  }
+
+  byte[] decompress(byte[] compressed, int originalLength) throws IOException {
+    Decompressor decompressor = mode.newDecompressor();
+    return decompress(decompressor, compressed, originalLength);
+  }
+
+  static byte[] decompress(Decompressor decompressor, byte[] compressed, int originalLength) throws IOException {
+    final BytesRef bytes = new BytesRef();
+    decompressor.decompress(new ByteArrayDataInput(compressed), originalLength, 0, originalLength, bytes);
+    return BytesRef.deepCopyOf(bytes).bytes;
+  }
+
+  byte[] decompress(byte[] compressed, int originalLength, int offset, int length) throws IOException {
+    Decompressor decompressor = mode.newDecompressor();
+    final BytesRef bytes = new BytesRef();
+    decompressor.decompress(new ByteArrayDataInput(compressed), originalLength, offset, length, bytes);
+    return BytesRef.deepCopyOf(bytes).bytes;
+  }
+
+  public void testDecompress() throws IOException {
+    final int iterations = atLeast(10);
+    for (int i = 0; i < iterations; ++i) {
```

Comment:
What's the main purpose to repeat this for 10 times?

Reply:
I  think this is for different segments. And I will remove the changes in this class because it was just for debugging convenience.